src/codegen/expression.rs

```diff
@@ -3060,3 +3121,21 @@ fn code(loc: &Loc, contract_no: usize, ns: &Namespace, opt: &Options) -> Express
 
     Expression::AllocDynamicBytes(*loc, Type::DynamicBytes, size.into(), Some(code))
 }
+
+pub(crate) fn report_error(
```

Comment:
This function adds a `Instr::DebugPrint {}`, so it should be called the same, e.g. `fn debug_print()` or so. Naming is in consistent - is it debug print or report error? There should be one name and the solang command line, `Instr` and this function should use the same term.

Also this needs to be called for `revert()`, `revert('this is my reason')`, and `require(expr, reason)` too.

Reply:
I think the name `Instr::ReportError` should be used instead of  `DebugPrint`, and the name of the function as well. thanks

for `revert`, there is this line: https://github.com/salaheldinsoliman/solang/blob/0353dbd3162174799b44c9fd6a3d81c7027b75d8/src/codegen/expression.rs#L1364 that if there is no expression within revert, it omits `revert encountered in line x`. So I will add the same for `require` as well

Reply:
Did you mean replacing Instr::Print with Instr::DebugPrint All together? @seanyoung

Reply:
> Did you mean replacing Instr::Print with Instr::DebugPrint All together? @seanyoung

I think that is an even better idea!

Reply:
Well, or replacing `Instr::DebugPrint` with `Instr::Print`. @xermicus  WDYT

Reply:
@seanyoung sorry somehow I missed this comment earlier on :facepalm: I agree, ideally this can be replaced with `Instr::Print`. Just glanced at the code, unless I missed something this should be easy. @salaheldinsoliman see also the [comment below](https://github.com/hyperledger/solang/pull/1150#discussion_r1099048822).